ush/python/pygfs/task/upp.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+
+import os
+from logging import getLogger
+from typing import Dict, Any, Union
+from pprint import pformat
+
+from pygw.attrdict import AttrDict
+from pygw.yaml_file import parse_j2yaml
+from pygw.file_utils import FileHandler
+from pygw.jinja import Jinja
+from pygw.logger import logit
+from pygw.task import Task
+from pygw.timetools import add_to_datetime, to_timedelta
+from pygw.exceptions import WorkflowException
+from pygw.executable import Executable, which
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class UPP(Task):
+    """Unified Post Processor Task
+    """
+
+    VALID_UPP_RUN = ['analysis', 'forecast', 'goes']
+
+    @logit(logger, name="UPP")
+    def __init__(self, config: Dict[str, Any]) -> None:
+        """Constructor for the UPP task
+        The constructor is responsible for resolving the "UPP_CONFIG" based in the run-type "upp_run"
+        Valid options for "upp_run" are:
+        analysis: process GSI analysis output
```

Comment:
just say 'analysis' since we should be able to use with JEDI (with tweaks to write out a gaussian anl nc file...)